src/store/options.js

```diff
@@ -75,8 +71,31 @@ const Options = {
 
 export default Options;
 
+export async function observe(property, fn) {
+  let value;
+  const wrapperFn = (options) => {
+    if (value === undefined || options[property] !== value) {
+      value = options[property];
+      fn(value);
+    }
+  };
+
+  // Add listener for options updates
+  chrome.runtime.onMessage.addListener((message) => {
+    if (message?.action === UPDATE_OPTIONS_ACTION_NAME) {
+      wrapperFn(message.options);
+    }
+
+    return false;
+  });
+
+  // Run for the first time with the current options
+  const options = await store.resolve(store.get(Options));
+  wrapperFn(options);
+}
+
 chrome.runtime.onMessage.addListener((message) => {
-  if (isUpdateOptionsMessage(message)) {
+  if (message?.action === UPDATE_OPTIONS_ACTION_NAME) {
```

Comment:
can a `message` be not an object? you don't check for that in other places

Reply:
It was there before, I think it might have been my precousion, we control the message passing, bit i am not sure If some of the feature might pass message, wchich is not an object (technically it is possible).

Reply:
As we don't check it in other places, I removed the operator.